warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -481,6 +481,22 @@ private DateIndexHelper getDateIndexHelper(final ShardQueryConfiguration config)
             }
         }
         
+        // Option to return query plans in random order for better distribution of running scans.
+        if (config.isPlanOrderRandom()) {
+            
+            if (queryPlanComparators == null)
+                queryPlanComparators = new ArrayList<>();
+            
+            queryPlanComparators.add(new DefaultQueryPlanComparator() {
+                @Override
+                public int compare(QueryPlan o1, QueryPlan o2) {
+                    int left = o1.getQueryString().hashCode() + o1.getRanges().iterator().next().getStartKey().hashCode();
+                    int right = o2.getQueryString().hashCode() + o2.getRanges().iterator().next().getStartKey().hashCode();
+                    return left - right;
+                }
+            });
```

Comment:
I love the idea of a random ordering of the ranges, but I do not like the idea that two identical queries would run in exactly the same order.  Check out this stack overflow post which gives an example of a way to create a uniquely random ordering using a comparator even if against the same input: https://stackoverflow.com/questions/29241746/shuffle-random-comparator

Reply:
Sounds reasonable